editor/src/clj/editor/gui.clj

```diff
@@ -331,47 +340,6 @@
 
 ;; /////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-(g/defnk produce-node-msg [type custom-type parent child-index _declared-properties]
-  (let [pb-renames {:x-anchor :xanchor
-                    :y-anchor :yanchor
-                    :generated-id :id}
-        v3-fields {:position [0.0 0.0 0.0] :scale [1.0 1.0 1.0] :size [200.0 100.0 0.0]}
-        props (:properties _declared-properties)
-        overridden-fields (->> props
-                               (keep (fn [[prop val]] (when (contains? val :original-value) (prop-key->prop-index prop))))
-                               (sort)
-                               (vec))
-        msg (-> {:type type
-                 :custom-type custom-type
-                 :child-index child-index
-                 :template-node-child false
-                 :overridden-fields overridden-fields}
-                (into (map (fn [[k v]] [k (:value v)])
-                           (filter (fn [[k v]] (and (get v :visible true)
-                                                    (not (contains? (set (keys pb-renames)) k))))
-                                   props)))
-                (cond->
-                 (and parent (not-empty parent)) (assoc :parent parent)
-                 (= type :type-template) (->
-                                          (update :template (fn [t] (resource/resource->proj-path (:resource t))))
-                                          (assoc :color [1.0 1.0 1.0 1.0]))
-                 (= type :type-particlefx) (->
-                                            (assoc
-                                             :size [1.0 1.0 0.0 1.0]
-                                             :size-mode :size-mode-auto)))
-                (into (map (fn [[k v]] [v (get-in props [k :value])]) pb-renames)))
-        msg (-> (reduce (fn [msg [k default]]
-                          (update msg k v3->v4 default))
-                        msg
-                        v3-fields)
-                (update :rotation (fn [rotation]
-                                    (clj-quat->euler-v4 (or rotation [0.0 0.0 0.0 1.0])))))
-        node-type-info (get-registered-node-type-info type custom-type)
-        msg (if-some [convert-fn (:convert-fn node-type-info)]
-              (convert-fn node-type-info msg)
-              msg)]
-    msg))
-
```

Comment:
Evaluating `_declared-properties` is expensive compared to explicitly depending on individual property values. We now have individual implementations of the `node-msg` output for each `GuiNode` type.